src/par/FixColumnPartitions.c

```diff
@@ -162,20 +186,38 @@ int FixColumnPartitions(Mesh_ptr mesh, int *part, MSTK_Comm comm) {
         curreg = nxtreg;
         curid = nxtid;
         topf = botf;
-      }
-      else 
+      } else {
         done = 1;
+      }
       List_Delete(fregs);
     }
-    
+
+    fprintf(fid, "%-d\n", num_cells_in_col);
   }
 
   if (nmove) {
     char msg[256];
-    sprintf(msg,"Redistributed %-d elements to maintain column partitioning",nmove);
+    sprintf(msg,"Redistributed %-d elements in %-d columns to maintain column partitioning",nmove, num_cols);
     MSTK_Report("FixColumnPartitions",msg,MSTK_MESG);
   }
-    
+
+  fclose(fid);
+  num_not_touched = 0;
```

Comment:
What if part of the mesh is columnar and the rest is not? Maybe the message should say that algorithm could not put all elements in a column - perhaps part of the mesh is not columnar?